hd_map: "simple_road_v1.osm"

global_planning:
  lanelet2_settings:
    TURNING_RADIUS: 0.3
    POSITION_TOLERANCE: 0.2
    YAW_TOLERANCE: 0.1

prediction:
  method: "constant_velocity_predictor"

local_planning:
  local_planning_frequency: 5
  lattice_settings:
    dt: 0.25 # time step
    D_T: 0.5 # prediction sampling length (s)
    MAX_T: 5.0 # max prediction time [s]
    MIN_T: 5.0 # min prediction time [s]

    d_r: 0.35
    d_l: 0.35
    D_S: 0.35 # sample Frenet d

    MAX_SPEED: 1 # maximum speed [m/s]
    MIN_SPEED: -0.25 # minimum speed [m/s]
    TARGET_SPEED: 0.5 # target speed [m/s]
    D_V_S: 0.25 # target speed sampling length [m/s]

    K_J: 0.1
    K_T: 0.5
    K_D: 5
    K_LAT: 0.1
    K_LON: 0.8

    MAX_ACCEL: 1 # maximum acceleration [m/ss]
    MAX_CURVATURE: 10.0 # maximum curvature [1/m]

control:
  control_frequency: 10
  pid_settings:
    lateral:
      K_P: 1.0
      K_I: 0.0
      K_D: 0.0
      output_max: 0.5
      output_min: -0.5
      dt: 0.1 
    longitudinal:
      K_P: 1.0
      K_I: 0.0
      K_D: 0.0
      output_max: 1
      output_min: 0
      dt: 0.1


vehicle_info:
  length: 0.35
  width: 0.22
